ml-agents-envs/mlagents/envs/side_channel/float_properties_channel.py

```diff
@@ -0,0 +1,77 @@
+from mlagents.envs.side_channel.side_channel import SideChannel, SideChannelType
+import struct
+from typing import Tuple, Optional, List
+
+
+class FloatPropertiesChannel(SideChannel):
+    """
+    This is the SideChannel for float properties shared with Unity.
+    You can modify the float properties of an environment with the commands
+    set_property, get_property and list_properties.
+    """
+
+    def __init__(self):
+        self._float_properties = {}
+        super().__init__()
+
+    @property
+    def channel_type(self) -> int:
+        return SideChannelType.FloatProperties
+
+    def on_message_received(self, data: bytearray) -> None:
+        """
+        Is called by the environment to the side channel. Can be called
+        multiple times per step if multiple messages are meant for that
+        SideChannel.
+        Note that Python should never receive an engine configuration from
+        Unity
+        """
+        k, v = self.deserialize_float_prop(data)
+        self._float_properties[k] = v
+
+    def set_property(self, key: str, value: float) -> None:
+        """
+        Sets a property in the Unity Environment.
+        :param key: The string identifier of the property.
+        :param value: The float value of the property.
+        """
+        self._float_properties[key] = value
+        super().queue_message_to_send(self.serialize_float_prop(key, value))
+
+    def get_property(self, key: str) -> Optional[float]:
+        """
+        Gets a property in the Unity Environment. If the property was not
+        found, will return None.
+        :param key: The string identifier of the property.
+        :return: The float value of the property or None.
+        """
+        if key in self._float_properties:
+            return self._float_properties[key]
+        else:
+            return None
```

Comment:
You can just do `self._float_properties.get(key)` - that returns None if the key isn't in the dict.